### ISSUE TITLE
Not recognizing established OAuth connection

### ISSUE THREAD

**Issue by ghost** [microsoft/botframework-solutions#2067]
#### What project is affected?

Calendar Skill OAuth Connections

#### What language is this in?

C#

#### What happens?

Cannot run the VA, throws exception.  I believe I have established the correct connections after the botskills connect was not able to do it automatically.  I used the az bot authsetting to create the connection after editing the AAD App Registration to no avail.

#### What are the steps to reproduce this issue?

This is the second VA I have created and both have run into this problem.  I am just following the documentation for connecting a skill to azure.  

#### What were you expecting to happen?

I expect the Startup.cs file to populate the settings.OAuthConnections with the correct OAuth connection but the value is null.  I even tried adding a connection to the skill itself and it does get entered correctly, but it is not the correct connection or location.  From my understanding it has to be a connection to AADV2 from the VA.  

#### Can you share any logs, error output, etc.?

'You must configure at least one supported OAuth connection to use this skill: Calendar Skill.'

#### Any screenshots or additional context?

![anno1](https://user-images.githubusercontent.com/51760832/62636695-0407ae80-b908-11e9-9271-b331473f8b22.png)
![anno2](https://user-images.githubusercontent.com/51760832/62636702-066a0880-b908-11e9-929a-a240cfbd69f8.png)

**Comment by ghost** [comment]
I tried to recreate the problem with the email skill and got this error.  
![anno3](https://user-images.githubusercontent.com/51760832/62706163-f3fdd680-b9bc-11e9-931c-4f9ca8adaccf.png)
with this log: 

Cannot find the entity extractor "Message.Any" in the specified application version.

LUIS Command Line Interface - ┬® 2018 Microsoft Corporation

Imports an application to LUIS, the application's structure should be included in in the request body.

Usage:
luis import application --in jSONApp.json --appName <string>

Command arguments are:
 --in (required)   The object to send in the body of the request                                                                                                                                                    
                   {                                                                                                                                                                                                
                     "intents": [                                                                                                                                                                                   
                       {                                                                                                                                                                                            
                         "name": "Greeting"                                                                                                                                                                         
                       }                                                                                                                                                                                            
                     ],                                                                                                                                                                                             
                     "entities": [                                                                                                                                                                                  
                       {                                                                                                                                                                                            
                         "name": "userName",                                                                                                                                                                        
                         "roles": []                                                                                                                                                                                
                       }                                                                                                                                                                                            
                     ],                                                                                                                                                                                             
                     "composites": [],                                                                                                                                                                              
                     "closedLists": [],                                                                                                                                                                             
                     "regex_entities": [],                                                                                                                                                                          
                     "model_features": [],                                                                                                                                                                          
                     "regex_features": [],                                                                                                                                                                          
                     "utterances": [                                                                                                                                                                                
                       {                                                                                                                                                                                            
                         "text": "Hi",                                                                                                                                                                              
                         "intent": "Greeting",                                                                                                                                                                      
                         "entities": []                                                                                                                                                                             
                       }                                                                                                                                                                                            
                     ],                                                                                                                                                                                             
                     "patterns": [],                                                                                                                                                                                
                     "patternAnyEntities": [],                                                                                                                                                                      
                     "prebuiltEntities": [],                                                                                                                                                                        
                     "luis_schema_version": "3.0.0",                                                                                                                                                                
                     "versionId": "0.1",                                                                                                                                                                            
                     "name": "test1",                                                                                                                                                                               
                     "desc": "",                                                                                                                                                                                    
                     "culture": "en-us"                                                                                                                                                                             
                   }                                                                                                                                                                                                
 --appName         The application name to create. If not specified, the application name will be read from the imported object.                                                                                    
 <string>                                                                                                                                                                                                           
 --msbot           (OPTIONAL) Format the output as json for piping into msbot connect luis command                                                                                                                  

Configuration and Overrides:
 --appId           Specifies the public LUIS application id. Overrides the .luisrc value and the LUIS_APP_ID environment variable.                                                                                  
 --authoringKey    Specifies the LUIS authoring  key (from luis.ai portal user settings page). Overrides the .luisrc value and the LUIS_AUTHORING_KEY environment variable.                                         
 --subscriptionKÔÇª  Specifies the LUIS subscriptionKey. Overrides the .luisrc value and the LUIS_SUBSCRIPTION_KEY environment variable.                                                                              
 --versionId       Specifies the version id. Overrides the .luisrc value and the LUIS_VERSION_ID environment variable.                                                                                              
 --region          Specifies the authoring region for all requests. [westus|westeurope|australiaeast] Overrides the .luisrc value and the LUIS_REGION environment variable.                                         
 --cloud           Specifies the cloud region for all requests. [com|us] Overrides the .luisrc value and the LUIS_CLOUD environment variable.                                                                       
 --stdin           Pull in service keys from stdin in the format of that is the output of: msbot get service                                                                                                        
 --prefix          Appends [luis-apis] prefix to all messages                                                                                                                                                       

Global Arguments:
 --help,    -h     Prints this help file.                                                                                                                                                                           
 --version, -v     Prints the version of this cli tool                                                                                                                                                              
 --force,   -f     Do not prompt for confirmation, force the operation                                                                                                                                              
 --!               Dumps all documented commands to the console with descriptions                                                                                                                                   

[31m Model                             [39m[31m Type               [39m[31m StatusId [39m[31m Status   [39m[31m          [39m
 Cancel                             Intent Classifier   9         Queued             
 Confirm                            Intent Classifier   9         Queued             
 DirectionalReference               Entity Extractor    9         Queued             
 Escalate                           Intent Classifier   9         Queued             
 FinishTask                         Intent Classifier   9         Queued             
 GoBack                             Intent Classifier   9         Queued             
 Help                               Intent Classifier   9         Queued             
 Logout                             Intent Classifier   9         Queued             
 None                               Intent Classifier   9         Queued             
 ReadAloud                          Intent Classifier   9         Queued             
 Reject                             Intent Classifier   9         Queued             
 Repeat                             Intent Classifier   9         Queued             
 SelectAny                          Intent Classifier   9         Queued             
 SelectItem                         Intent Classifier   9         Queued             
 SelectNone                         Intent Classifier   9         Queued             
 ShowNext                           Intent Classifier   9         Queued             
 ShowPrevious                       Intent Classifier   9         Queued             
 StartOver                          Intent Classifier   9         Queued             
 Stop                               Intent Classifier   9         Queued             
[31m Model                             [39m[31m Type               [39m[31m StatusId [39m[31m Status   [39m[31m          [39m
 Cancel                             Intent Classifier   3         InProgrÔÇª           
 Confirm                            Intent Classifier   3         InProgrÔÇª           
 DirectionalReference               Entity Extractor    3         InProgrÔÇª           
 Escalate                           Intent Classifier   3         InProgrÔÇª           
 FinishTask                         Intent Classifier   3         InProgrÔÇª           
 GoBack                             Intent Classifier   3         InProgrÔÇª           
 Help                               Intent Classifier   3         InProgrÔÇª           
 Logout                             Intent Classifier   3         InProgrÔÇª           
 None                               Intent Classifier   3         InProgrÔÇª           
 ReadAloud                          Intent Classifier   3         InProgrÔÇª           
 Reject                             Intent Classifier   3         InProgrÔÇª           
 Repeat                             Intent Classifier   3         InProgrÔÇª           
 SelectAny                          Intent Classifier   3         InProgrÔÇª           
 SelectItem                         Intent Classifier   3         InProgrÔÇª           
 SelectNone                         Intent Classifier   3         InProgrÔÇª           
 ShowNext                           Intent Classifier   3         InProgrÔÇª           
 ShowPrevious                       Intent Classifier   3         InProgrÔÇª           
 StartOver                          Intent Classifier   3         InProgrÔÇª           
 Stop                               Intent Classifier   3         InProgrÔÇª           
[31m Model                             [39m[31m Type               [39m[31m StatusId [39m[31m Status   [39m[31m          [39m
 Cancel                             Intent Classifier   3         InProgrÔÇª           
 Confirm                            Intent Classifier   3         InProgrÔÇª           
 DirectionalReference               Entity Extractor    3         InProgrÔÇª           
 Escalate                           Intent Classifier   3         InProgrÔÇª           
 FinishTask                         Intent Classifier   3         InProgrÔÇª           
 GoBack                             Intent Classifier   3         InProgrÔÇª           
 Help                               Intent Classifier   3         InProgrÔÇª           
 Logout                             Intent Classifier   3         InProgrÔÇª           
 None                               Intent Classifier   3         InProgrÔÇª           
 ReadAloud                          Intent Classifier   3         InProgrÔÇª           
 Reject                             Intent Classifier   3         InProgrÔÇª           
 Repeat                             Intent Classifier   3         InProgrÔÇª           
 SelectAny                          Intent Classifier   3         InProgrÔÇª           
 SelectItem                         Intent Classifier   3         InProgrÔÇª           
 SelectNone                         Intent Classifier   3         InProgrÔÇª           
 ShowNext                           Intent Classifier   3         InProgrÔÇª           
 ShowPrevious                       Intent Classifier   3         InProgrÔÇª           
 StartOver                          Intent Classifier   3         InProgrÔÇª           
 Stop                               Intent Classifier   3         InProgrÔÇª           
[31m Model                             [39m[31m Type               [39m[31m StatusId [39m[31m Status   [39m[31m          [39m
 Cancel                             Intent Classifier   3         InProgrÔÇª           
 Confirm                            Intent Classifier   3         InProgrÔÇª           
 DirectionalReference               Entity Extractor    3         InProgrÔÇª           
 Escalate                           Intent Classifier   3         InProgrÔÇª           
 FinishTask                         Intent Classifier   3         InProgrÔÇª           
 GoBack                             Intent Classifier   3         InProgrÔÇª           
 Help                               Intent Classifier   3         InProgrÔÇª           
 Logout                             Intent Classifier   3         InProgrÔÇª           
 None                               Intent Classifier   3         InProgrÔÇª           
 ReadAloud                          Intent Classifier   3         InProgrÔÇª           
 Reject                             Intent Classifier   3         InProgrÔÇª           
 Repeat                             Intent Classifier   3         InProgrÔÇª           
 SelectAny                          Intent Classifier   3         InProgrÔÇª           
 SelectItem                         Intent Classifier   3         InProgrÔÇª           
 SelectNone                         Intent Classifier   3         InProgrÔÇª           
 ShowNext                           Intent Classifier   3         InProgrÔÇª           
 ShowPrevious                       Intent Classifier   3         InProgrÔÇª           
 StartOver                          Intent Classifier   3         InProgrÔÇª           
 Stop                               Intent Classifier   3         InProgrÔÇª           
[31m Model                             [39m[31m Type               [39m[31m StatusId [39m[31m Status   [39m[31m          [39m
 Cancel                             Intent Classifier   3         InProgrÔÇª           
 Confirm                            Intent Classifier   3         InProgrÔÇª           
 DirectionalReference               Entity Extractor    3         InProgrÔÇª           
 Escalate                           Intent Classifier   3         InProgrÔÇª           
 FinishTask                         Intent Classifier   3         InProgrÔÇª           
 GoBack                             Intent Classifier   3         InProgrÔÇª           
 Help                               Intent Classifier   3         InProgrÔÇª           
 Logout                             Intent Classifier   3         InProgrÔÇª           
 None                               Intent Classifier   3         InProgrÔÇª           
 ReadAloud                          Intent Classifier   3         InProgrÔÇª           
 Reject                             Intent Classifier   3         InProgrÔÇª           
 Repeat                             Intent Classifier   3         InProgrÔÇª           
 SelectAny                          Intent Classifier   3         InProgrÔÇª           
 SelectItem                         Intent Classifier   3         InProgrÔÇª           
 SelectNone                         Intent Classifier   3         InProgrÔÇª           
 ShowNext                           Intent Classifier   3         InProgrÔÇª           
 ShowPrevious                       Intent Classifier   3         InProgrÔÇª           
 StartOver                          Intent Classifier   3         InProgrÔÇª           
 Stop                               Intent Classifier   3         InProgrÔÇª           
[31m Model                             [39m[31m Type               [39m[31m StatusId [39m[31m Status   [39m[31m          [39m
 Cancel                             Intent Classifier   3         InProgrÔÇª           
 Confirm                            Intent Classifier   3         InProgrÔÇª           
 DirectionalReference               Entity Extractor    3         InProgrÔÇª           
 Escalate                           Intent Classifier   3         InProgrÔÇª           
 FinishTask                         Intent Classifier   3         InProgrÔÇª           
 GoBack                             Intent Classifier   3         InProgrÔÇª           
 Help                               Intent Classifier   3         InProgrÔÇª           
 Logout                             Intent Classifier   3         InProgrÔÇª           
 None                               Intent Classifier   3         InProgrÔÇª           
 ReadAloud                          Intent Classifier   3         InProgrÔÇª           
 Reject                             Intent Classifier   3         InProgrÔÇª           
 Repeat                             Intent Classifier   3         InProgrÔÇª           
 SelectAny                          Intent Classifier   3         InProgrÔÇª           
 SelectItem                         Intent Classifier   3         InProgrÔÇª           
 SelectNone                         Intent Classifier   3         InProgrÔÇª           
 ShowNext                           Intent Classifier   3         InProgrÔÇª           
 ShowPrevious                       Intent Classifier   3         InProgrÔÇª           
 StartOver                          Intent Classifier   3         InProgrÔÇª           
 Stop                               Intent Classifier   3         InProgrÔÇª           
[31m Model                             [39m[31m Type               [39m[31m StatusId [39m[31m Status   [39m[31m          [39m
 Cancel                             Intent Classifier   3         InProgrÔÇª           
 Confirm                            Intent Classifier   3         InProgrÔÇª           
 DirectionalReference               Entity Extractor    3         InProgrÔÇª           
 Escalate                           Intent Classifier   3         InProgrÔÇª           
 FinishTask                         Intent Classifier   3         InProgrÔÇª           
 GoBack                             Intent Classifier   3         InProgrÔÇª           
 Help                               Intent Classifier   3         InProgrÔÇª           
 Logout                             Intent Classifier   3         InProgrÔÇª           
 None                               Intent Classifier   3         InProgrÔÇª           
 ReadAloud                          Intent Classifier   3         InProgrÔÇª           
 Reject                             Intent Classifier   3         InProgrÔÇª           
 Repeat                             Intent Classifier   3         InProgrÔÇª           
 SelectAny                          Intent Classifier   3         InProgrÔÇª           
 SelectItem                         Intent Classifier   3         InProgrÔÇª           
 SelectNone                         Intent Classifier   3         InProgrÔÇª           
 ShowNext                           Intent Classifier   3         InProgrÔÇª           
 ShowPrevious                       Intent Classifier   3         InProgrÔÇª           
 StartOver                          Intent Classifier   3         InProgrÔÇª           
 Stop                               Intent Classifier   3         InProgrÔÇª           
[31m Model                             [39m[31m Type               [39m[31m StatusId [39m[31m Status   [39m[31m          [39m
 Cancel                             Intent Classifier   3         InProgrÔÇª           
 Confirm                            Intent Classifier   3         InProgrÔÇª           
 DirectionalReference               Entity Extractor    3         InProgrÔÇª           
 Escalate                           Intent Classifier   3         InProgrÔÇª           
 FinishTask                         Intent Classifier   3         InProgrÔÇª           
 GoBack                             Intent Classifier   3         InProgrÔÇª           
 Help                               Intent Classifier   3         InProgrÔÇª           
 Logout                             Intent Classifier   3         InProgrÔÇª           
 None                               Intent Classifier   3         InProgrÔÇª           
 ReadAloud                          Intent Classifier   3         InProgrÔÇª           
 Reject                             Intent Classifier   3         InProgrÔÇª           
 Repeat                             Intent Classifier   3         InProgrÔÇª           
 SelectAny                          Intent Classifier   3         InProgrÔÇª           
 SelectItem                         Intent Classifier   3         InProgrÔÇª           
 SelectNone                         Intent Classifier   3         InProgrÔÇª           
 ShowNext                           Intent Classifier   3         InProgrÔÇª           
 ShowPrevious                       Intent Classifier   3         InProgrÔÇª           
 StartOver                          Intent Classifier   3         InProgrÔÇª           
 Stop                               Intent Classifier   3         InProgrÔÇª           
[31m Model                             [39m[31m Type               [39m[31m StatusId [39m[31m Status   [39m[31m          [39m
 Cancel                             Intent Classifier   3         InProgrÔÇª           
 Confirm                            Intent Classifier   3         InProgrÔÇª           
 DirectionalReference               Entity Extractor    3         InProgrÔÇª           
 Escalate                           Intent Classifier   3         InProgrÔÇª           
 FinishTask                         Intent Classifier   3         InProgrÔÇª           
 GoBack                             Intent Classifier   3         InProgrÔÇª           
 Help                               Intent Classifier   3         InProgrÔÇª           
 Logout                             Intent Classifier   3         InProgrÔÇª           
 None                               Intent Classifier   3         InProgrÔÇª           
 ReadAloud                          Intent Classifier   3         InProgrÔÇª           
 Reject                             Intent Classifier   3         InProgrÔÇª           
 Repeat                             Intent Classifier   3         InProgrÔÇª           
 SelectAny                          Intent Classifier   3         InProgrÔÇª           
 SelectItem                         Intent Classifier   3         InProgrÔÇª           
 SelectNone                         Intent Classifier   3         InProgrÔÇª           
 ShowNext                           Intent Classifier   3         InProgrÔÇª           
 ShowPrevious                       Intent Classifier   3         InProgrÔÇª           
 StartOver                          Intent Classifier   3         InProgrÔÇª           
 Stop                               Intent Classifier   3         InProgrÔÇª           
[31m Model                             [39m[31m Type               [39m[31m StatusId [39m[31m Status   [39m[31m          [39m
 Cancel                             Intent Classifier   3         InProgrÔÇª           
 Confirm                            Intent Classifier   3         InProgrÔÇª           
 DirectionalReference               Entity Extractor    3         InProgrÔÇª           
 Escalate                           Intent Classifier   3         InProgrÔÇª           
 FinishTask                         Intent Classifier   3         InProgrÔÇª           
 GoBack                             Intent Classifier   3         InProgrÔÇª           
 Help                               Intent Classifier   3         InProgrÔÇª           
 Logout                             Intent Classifier   3         InProgrÔÇª           
 None                               Intent Classifier   3         InProgrÔÇª           
 ReadAloud                          Intent Classifier   3         InProgrÔÇª           
 Reject                             Intent Classifier   3         InProgrÔÇª           
 Repeat                             Intent Classifier   3         InProgrÔÇª           
 SelectAny                          Intent Classifier   3         InProgrÔÇª           
 SelectItem                         Intent Classifier   3         InProgrÔÇª           
 SelectNone                         Intent Classifier   3         InProgrÔÇª           
 ShowNext                           Intent Classifier   3         InProgrÔÇª           
 ShowPrevious                       Intent Classifier   3         InProgrÔÇª           
 StartOver                          Intent Classifier   3         InProgrÔÇª           
 Stop                               Intent Classifier   3         InProgrÔÇª           
[31m Model                             [39m[31m Type               [39m[31m StatusId [39m[31m Status   [39m[31m          [39m
 Cancel                             Intent Classifier   3         InProgrÔÇª           
 Confirm                            Intent Classifier   3         InProgrÔÇª           
 DirectionalReference               Entity Extractor    3         InProgrÔÇª           
 Escalate                           Intent Classifier   3         InProgrÔÇª           
 FinishTask                         Intent Classifier   3         InProgrÔÇª           
 GoBack                             Intent Classifier   3         InProgrÔÇª           
 Help                               Intent Classifier   3         InProgrÔÇª           
 Logout                             Intent Classifier   3         InProgrÔÇª           
 None                               Intent Classifier   3         InProgrÔÇª           
 ReadAloud                          Intent Classifier   3         InProgrÔÇª           
 Reject                             Intent Classifier   3         InProgrÔÇª           
 Repeat                             Intent Classifier   3         InProgrÔÇª           
 SelectAny                          Intent Classifier   3         InProgrÔÇª           
 SelectItem                         Intent Classifier   3         InProgrÔÇª           
 SelectNone                         Intent Classifier   3         InProgrÔÇª           
 ShowNext                           Intent Classifier   3         InProgrÔÇª           
 ShowPrevious                       Intent Classifier   3         InProgrÔÇª           
 StartOver                          Intent Classifier   3         InProgrÔÇª           
 Stop                               Intent Classifier   3         InProgrÔÇª           
[31m Model                             [39m[31m Type               [39m[31m StatusId [39m[31m Status   [39m[31m          [39m
 Cancel                             Intent Classifier   3         InProgrÔÇª           
 Confirm                            Intent Classifier   3         InProgrÔÇª           
 DirectionalReference               Entity Extractor    3         InProgrÔÇª           
 Escalate                           Intent Classifier   3         InProgrÔÇª           
 FinishTask                         Intent Classifier   3         InProgrÔÇª           
 GoBack                             Intent Classifier   3         InProgrÔÇª           
 Help                               Intent Classifier   3         InProgrÔÇª           
 Logout                             Intent Classifier   3         InProgrÔÇª           
 None                               Intent Classifier   3         InProgrÔÇª           
 ReadAloud                          Intent Classifier   3         InProgrÔÇª           
 Reject                             Intent Classifier   3         InProgrÔÇª           
 Repeat                             Intent Classifier   3         InProgrÔÇª           
 SelectAny                          Intent Classifier   3         InProgrÔÇª           
 SelectItem                         Intent Classifier   3         InProgrÔÇª           
 SelectNone                         Intent Classifier   3         InProgrÔÇª           
 ShowNext                           Intent Classifier   3         InProgrÔÇª           
 ShowPrevious                       Intent Classifier   3         InProgrÔÇª           
 StartOver                          Intent Classifier   3         InProgrÔÇª           
 Stop                               Intent Classifier   3         InProgrÔÇª           
[31m Model                             [39m[31m Type               [39m[31m StatusId [39m[31m Status   [39m[31m          [39m
 Cancel                             Intent Classifier   0         Success            
 Confirm                            Intent Classifier   0         Success            
 Escalate                           Intent Classifier   0         Success            
 FinishTask                         Intent Classifier   0         Success            
 GoBack                             Intent Classifier   0         Success            
 Help                               Intent Classifier   0         Success            
 Logout                             Intent Classifier   0         Success            
 None                               Intent Classifier   0         Success            
 ReadAloud                          Intent Classifier   0         Success            
 Reject                             Intent Classifier   0         Success            
 Repeat                             Intent Classifier   0         Success            
 SelectAny                          Intent Classifier   0         Success            
 SelectItem                         Intent Classifier   0         Success            
 SelectNone                         Intent Classifier   0         Success            
 ShowNext                           Intent Classifier   0         Success            
 ShowPrevious                       Intent Classifier   0         Success            
 StartOver                          Intent Classifier   0         Success            
 Stop                               Intent Classifier   0         Success            
 DirectionalReference               Entity Extractor    0         Success

**Comment by anusharr** [comment]
@tjyml15 did you test and see if your OAuth connection is configured correctly in the portal? In the meanwhile, could you elaborate on the steps you followed or link the documentation which you referred to so that I can repro your issue?

**Comment by darrenj** [comment]
I can repro the Message.Any issue with the Email skill.

![image](https://user-images.githubusercontent.com/33519520/63181278-86e6e280-c047-11e9-921b-3dd96d176214.png)

```

Cannot find the entity extractor "Message.Any" in the specified application version.

LUIS Command Line Interface - ┬® 2018 Microsoft Corporation

Imports an application to LUIS, the application's structure should be included in in the request body.

Usage:
luis import application --in jSONApp.json --appName <string>

Command arguments are:
 --in (required)   The object to send in the body of the request                                                                                                                      
```

**Comment by DingmaomaoBJTU** [comment]
Yes, this is a known issue as ludown v1.3.2 failed to parse pattern entities. https://github.com/microsoft/botbuilder-tools/issues/1225. We can expect it fixed in v1.3.3. Currently to make a work around, please use npm install -g ludown@1.3.1 to downgrade.
